feat(analytics): track action buttons on Paper, Multimedia, and Exam Details (#636)

Matinj031 opened the pull request:
This PR implements end-to-end click tracking for action buttons across key detail pages and wires events to GA4 via Google Tag Manager.

* Adds unique data-action-id to each actionable button
* Pushes standardized action_click events with rich context to the dataLayer
* Provides a reusable tracking composable and a global click handler
* Initializes GTM via env-driven container ID

### Linked issue: #636 Track action buttons on Paper, Multimedia, and Exam Details pages

Matinj031 committed:
feat(exam details): show Related Past Papers on exam details (#626)

Result: Users see a “Related Past Papers” carousel under exam details (hidden when none).

Matinj031 committed:
feat(nuxt.config): add Google Tag Manager ID configuration

Matinj031 committed:
feat(env): add Google Tag Manager ID to example environment configuration

Matinj031 committed:
feat(exam details): enhance tracking for exam actions and add data attributes for analytics

Matinj031 committed:
feat(detail sidebar): add download tracking and data attribute for analytics

Matinj031 committed:
feat(content info): add data attributes for download buttons and implement tracking for download actions

Matinj031 committed:
feat(gtm tracking): implement composable for Google Tag Manager tracking of action button clicks, downloads, payments, and quiz starts

Matinj031 committed:
feat(gtm tracking): add GTM client plugin for script injection and action tracking on button clicks

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

Matinj031 committed:
feat(HeroSection): enhance token price display with loading state and price change indication

Matinj031 committed:
refactor: remove GTM plugins and composable

- Remove plugins/gtm.client.ts (GTM initialization)
- Remove plugins/gtm-action-tracking.client.js (GTM action tracking)
- Remove composables/useGtmTracking.ts (GTM tracking composable)

These are replaced with direct GA4 tracking for cleaner implementation.

Matinj031 committed:
refactor: remove GTM plugins and composable

- Remove plugins/gtm.client.ts (GTM initialization)
- Remove plugins/gtm-action-tracking.client.js (GTM action tracking)
- Remove composables/useGtmTracking.ts (GTM tracking composable)

These are replaced with direct GA4 tracking for cleaner implementation.

Matinj031 committed:
refactor: remove GA4 plugins for performance optimization

- Remove plugins/ga4.client.js (GA4 initialization plugin)
- Remove plugins/ga4-events.client.js (GA4 events tracking plugin)

Per frontend team lead recommendation to avoid plugins for better performance.
GA4 tracking will be handled directly in Google Analytics website instead.

Matinj031 committed:
feat: add GA4 script directly to HTML head

- Add Google Analytics 4 script to nuxt.config.js app.head.script
- Use Measurement ID: G-F4X87M3W8P
- Load gtag script asynchronously for better performance
- Initialize GA4 tracking without plugins

This approach avoids plugin performance issues while maintaining GA4 tracking.

Matinj031 committed:
feat: add Google Tag Manager script to nuxt.config.js

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

Matinj031 committed:
feat: integrate Google Tag Manager and Google Analytics configuration in nuxt.config.js

- Add GTM and GA4 IDs from environment variables for tracking
- Include necessary modules for GTM and GA4
- Configure GTM and GA4 settings for improved tracking and performance

Matinj031 committed:
chore: update environment configuration and package dependencies

- Add NUXT_PUBLIC_GTAG_ID to .env.example for Google Tag Manager integration
- Update package.json and package-lock.json to include @zadigetvoltaire/nuxt-gtm package for GTM support

Matinj031 committed:
feat(gtag): enable Google Analytics tracking in production environment

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

sanaderi commented on package.json:
```diff
@@ -67,6 +67,7 @@
     "@nuxt/eslint-config": "^1.6.0",
     "@types/node": "22.12.0",
     "@vite-pwa/nuxt": "^1.0.4",
+    "@zadigetvoltaire/nuxt-gtm": "^0.0.13",
```
Plz use official plugin 
https://www.npmjs.com/package/@nuxtjs/gtm

Matinj031 committed:
feat(gtm): implement file download tracking across components

- Removed unused data-action-id attributes from SidebarDetails, DetailSidebar, and ContentInfo components.
- Integrated trackFileDownload function from useGtmEvents composable to track file downloads for multimedia, past papers, and exams.
- Updated nuxt.config.js to enable GTM debug mode and load script for improved tracking.

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

Matinj031 committed:
Merge branch 'upgrade-v3' of https://github.com/GamaEdtech/front into feat/trackActionButtons

Matinj031 committed:
chore: remove nuxt-gtag dependency from configuration and package files

sanaderi commented on .env.example:
```diff
@@ -18,6 +18,12 @@ NUXT_RECAPTCHA_SITE_KEY=6LcGF3ErAAAAAAneMnvDNOvdBg4Z7IDoL86tJr9T
 NUXT_GOOGLE_ADSENSE_ID='ca-pub-3940256099942544'
 
+# Google Tag Manager Id
+NUXT_PUBLIC_GTAG_ID=
```
Plz Remove this line

sanaderi commented on nuxt.config.js:
```diff
@@ -12,17 +12,19 @@ export default defineNuxtConfig({
       solanaNetwork: process.env.NUXT_SOLANA_NETWORK,
       randomCoinApiKey: process.env.NUXT_PUBLIC_RANDOM_COIN_API_KEY,
+      gtmId: process.env.NUXT_PUBLIC_GTM_ID,
+      gtagId: process.env.NUXT_PUBLIC_GTAG_ID,
```
Plz remove this line

sanaderi commented on pages/exam/[id]/[[slug]].vue:
```diff
@@ -257,6 +272,28 @@ function updateGalleryData() {
 }
 
+// Check for related pastpapers to decide whether to render the section
+async function checkRelatedPastPapers() {
```
Remove this function

Matinj031 committed:
chore: clean up unused variables and configurations in exam page and environment files
